qirlib/src/codegen.rs

```diff
@@ -1,111 +1,64 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use inkwell::OptimizationLevel;
-
 use std::path::Path;
 
 use crate::{
     constants::Constants, intrinsics::Intrinsics, module, runtime_library::RuntimeLibrary,
     types::Types,
 };
 
-pub struct BareContext<'ctx> {
+pub struct CodeGenerator<'ctx> {
     pub context: &'ctx inkwell::context::Context,
     pub module: inkwell::module::Module<'ctx>,
     pub builder: inkwell::builder::Builder<'ctx>,
-}
-
-impl<'ctx> BareContext<'ctx> {
-    pub fn new(
-        context: &'ctx inkwell::context::Context,
-        context_type: ContextType<'ctx>,
-    ) -> Result<Self, String> {
-        let builder = context.create_builder();
-        let module = module::load_module(context, context_type)?;
-        Ok(BareContext {
-            builder,
-            module,
-            context,
-        })
-    }
-}
-
-pub struct Context<'ctx> {
-    pub context: &'ctx inkwell::context::Context,
-    pub module: inkwell::module::Module<'ctx>,
-    #[cfg(feature = "jit")]
-    pub execution_engine: inkwell::execution_engine::ExecutionEngine<'ctx>,
-    pub builder: inkwell::builder::Builder<'ctx>,
     pub types: Types<'ctx>,
     pub runtime_library: RuntimeLibrary<'ctx>,
     pub intrinsics: Intrinsics<'ctx>,
     pub constants: Constants<'ctx>,
 }
 
-pub enum ContextType<'ctx> {
+#[derive(Clone, Copy)]
+pub enum ModuleSource<'ctx> {
```

Comment:
Should ModuleSource be moved to qirlib::module?

Reply:
@samarsha Done.